### DISCUSSION TITLE
Is it normal to have dozens of `<div data-portal="true">` rendered?

### DISCUSSION THREAD

**Discussion by chris-m92** [mantinedev/mantine#6081] [Q&A] [answered]
So far I have been enjoying Mantine - it was a big change from MUI but a welcome one.

One thing I have noticed, however, is when looking at the DOM once the site is painted, there are dozens of `div` elements with the prop `data-portal="true"` rendered at the end of the `document`. Is this normal? My assumption is that because many menus, selects, and other components use the `withinPortal` by default and this is how it ends up rendering, but it just seems excessive.

Basically, the question is. Should I really need to "worry" about this?

![image](https://github.com/mantinedev/mantine/assets/45608364/f45f2caa-5e28-4ec0-937f-87c7ed45b81c)

**Comment by rtivital** [comment, marked as answer]
Yes, it is normal, this is how Portal element works. If you would like to avoid this, you can create a single container element where each element will go. To do that, use `target` prop in default props for Portal component.
